src/Mobs/PathFinder.h

```diff
@@ -0,0 +1,92 @@
+
+#pragma once
+
+#include "Path.h"
+
+/*
+TODO DOXY style
+
+This class wraps cPath.
+cPath is a "dumb device" - You give it point A and point B, and it returns a full path path.
+cPathFinder - You give it a constant stream of point A (where you are) and point B (where you want to go),
+and it tells you where to go next. It manages path recalculation internally, and is much more efficient that calling cPath every step.
+
+*/
+
+class cPathFinder
+{
+
+public:
+	/** Creates a cPathFinder instance. Each mob should have one cPathFinder throughout its lifetime.
+	@param a_MobWidth The mob width.
+	@param a_MobWidth The mob height.
+	*/
+	cPathFinder(double a_MobWidth, double a_MobHeight);
+
+	/** Updates the PathFinder's internal state and returns a waypoint.
+	A waypoint is a coordinate which the mob can safely move to from its current position in a straight line.
+	The mob is expected to call this function tick as long as it is following a path.
+	@param a_Chunk The chunk in which the mob is currently at.
+	@param a_Source The mob's position. a_Source's coordinates are expected to be within the chunk given in a_Chunk.
+	@param a_Destination The position the mob would like to reach,
+	@param a_OutputWaypoint An output parameter: The next waypoint to go to.
+
+	Returns an ePathFinderStatus.
+	ePathFinderStatus:CALCULATING - The PathFinder is still processing a path. Nothing was written to a_OutputWaypoint. The mob should probably not move.
+	ePathFinderStatus:PATH_FOUND - The PathFinder has found a path to the target. The next waypoint was written a_OutputWaypoint. The mob should probably move to a_OutputWaypoint.
+	ePathFinderStatus:NEARBY_FOUND - The PathFinder did not find a destination to the target but did find a nearby spot. The next waypoint was written a_OutputWaypoint. The mob should probably move to a_OutputWaypoint.
+	ePathFinderStatus:PATH_NOT_FOUND - The PathFinder did not find a destination to the target. Nothing was written to a_OutputWaypoint. The mob should probably not move.
+
+	Note: Once NEARBY_FOUND is returned once, subsequent calls return PATH_FOUND. This may be changed in the future. */
+	ePathFinderStatus GetNextWayPoint(cChunk & a_Chunk, Vector3d a_Source, Vector3d a_Destination, Vector3d & a_OutputWaypoint);
+
+private:
+
+	/** The width of the Mob which owns this PathFinder. */
+	double m_Width;
+
+	/** The height of the Mob which owns this PathFinder. */
+	double m_Height;
+
+	/** The current cPath instance we have. This is discarded and recreated when a path recalculation is needed. */
+	cPath *  m_Path;  // TODO unique ptr
```

Comment:
I thought so too. It seems a pointer is needed for line 57, but I'll rewrite that.